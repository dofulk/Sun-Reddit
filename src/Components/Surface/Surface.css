.Surface {
  background-color: rgb(1, 112, 112);
  border-radius: 2px;
  display: inline-flex;
  box-shadow: 2px 2px;
  color: white;
  flex-wrap: wrap;
  flex-direction: row;
  margin: 30px;
  margin-bottom: 110px;
}


@media only screen and (max-width: 768px) {
  .Surface {
    flex-direction: column;
  }
  .Comments {
    margin-left: 0px;
  }
}
.Content {
  flex: 3;
  margin: 10px
}


.Comments {
  flex: 2;
  display: flex;
  flex-direction: column;
  margin: 10px;
  max-height: 80vh;
  overflow: scroll;
}

.Comment-list {
  list-style-type: none;
  display: flex;
  flex-direction: column;
  list-style-position: outside;
  flex-grow: 1
}

.Input {
  justify-self: flex-end;
}