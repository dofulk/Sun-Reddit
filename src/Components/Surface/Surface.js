import React from 'react';
import './Surface.css';
import CommentList from '../CommentList/CommentList';
import Box from '@material-ui/core/Box';

const Surface = (props) => {


  return (
    <div className="Surface">
        <h1>{props.title}</h1>
        <Box className="Body">
          {props.body}
        </Box>
      <Box className="Comments">
        <CommentList comments={props.comments || []} overflow="auto"/>
      </Box>
    </div>
  )
}

export default Surface