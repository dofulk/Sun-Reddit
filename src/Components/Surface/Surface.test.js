import React from 'react';
import renderer from 'react-test-renderer';
import Surface from './Surface';
import Cat from '../../cateyes.jpg'

it('renders without props', () => {
  const tree = renderer.create(<Surface />).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders with image', () => {
  const tree = renderer.create(<Surface title="Test" body={Cat} comments={[
    {
      key: 1,
      comment: 'hi'
    },
    {
      key: 2,
      comment: 'hello',
    },
    {
      key: 3,
      comment: 'how'
    }]}/>).toJSON();
  expect(tree).toMatchSnapshot();
})