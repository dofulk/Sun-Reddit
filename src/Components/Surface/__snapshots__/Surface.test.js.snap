// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with image 1`] = `
<div
  className="Surface"
>
  <h1>
    Test
  </h1>
  <div
    className="MuiBox-root MuiBox-root-8 Body"
  >
    cateyes.jpg
  </div>
  <div
    className="MuiBox-root MuiBox-root-9 Comments"
  >
    <ul
      className="MuiList-root MuiList-padding"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            hi
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            hello
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
          >
            how
          </span>
        </div>
      </div>
      <hr
        className="MuiDivider-root"
      />
    </ul>
  </div>
</div>
`;

exports[`renders without props 1`] = `
<div
  className="Surface"
>
  <h1 />
  <div
    className="MuiBox-root MuiBox-root-2 Body"
  />
  <div
    className="MuiBox-root MuiBox-root-3 Comments"
  >
    <ul
      className="MuiList-root MuiList-padding"
    />
  </div>
</div>
`;
